Make networking panels pod matchers work with helm chart

dimitarvdimitrov opened the pull request:
<!--  Thanks for sending a pull request!  Before submitting:

1. Read our CONTRIBUTING.md guide
2. Rebase your PR if it gets out of sync with main
-->

#### What this PR does

In similar spirit as https://github.com/grafana/mimir/pull/1913

The pods created by the helm chart follow a format of
`<helm_release_name>-mimir-<ingester|distributor|...>`.

This is a problem for all places that use the per_instance_label for
matching. The per_instance_label is mostly used in aggregations (sum by
(pod), count by (pod), ...). The networking panels are the only ones
that use it for matching.

Signed-off-by: Dimitar Dimitrov <dimitar.dimitrov@grafana.com>


#### Which issue(s) this PR fixes or relates to

Fixes #<issue number>

#### Checklist

- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

dimitarvdimitrov commented on operations/mimir-mixin/dashboards/dashboard-utils.libsonnet:
```diff
@@ -233,7 +233,7 @@ local utils = import 'mixin-utils/utils.libsonnet';
     $.panel(title) +
     $.queryPanel(
-      'sum by(%(instance)s) (rate(%(metric)s{%(namespace)s,%(instance)s=~"%(instanceName)s"}[$__rate_interval]))' % {
+      'sum by(%(instance)s) (rate(%(metric)s{%(namespace)s,%(instance)s=~".*%(instanceName)s"}[$__rate_interval]))' % {
```
The chart allows to override both the `<release_name>-mimir-` part or only the `-mimir-` part.

We can choose to only support the default variant of the current version of the chart: `<release_name>-mimir-<component>`.

WDYT about adding another config option with the helm chart prefix? By default it will be `.*-mimir-` to match the chart default. Otherwise, folks can override it if they've chosen to override the names in the chart. I think this is the direction in which the chart went. I'm slightly against it because it gives too much customizability (google thinks that's not a word). But on the other hand it may be better to keep some parity between the jsonnet and the chart.

pracucci replied:
I 💯  agree with your point. Ideally dashboards should just work out of the box. Make it customizable is a solution (and I'm up to it). The only downside is that if the default config doesn't fit your use case, then you will have to change the config and compile the dashboards (not that it's difficult, it's just a command, but adds friction).

Can `<release_name>` include `-`? If not, then we could have a regex which matches `<word>-<word>-<component>` for the Helm case.

dimitarvdimitrov replied:
I pushed changes to the regex you proposed, waiting on opinion(s) for adding a config option.

dimitarvdimitrov replied:
oh, wow. github again showing its caching functions. I didn't see your comment when I posted mine. and I _am sure_ I refreshed before posting it

dimitarvdimitrov replied:
> Can `<release_name>` include `-`? 

yes, anything that's a valid k8s name is a valid release name

pracucci reviewed (approved):
I think it's a good starting point. I'm wondering if we can make the `-mimir-` part **not** configurable in Helm (what's the use case to configure it?). Otherwise, we should make it configurable in our mixin.

Please add a CHANGELOG entry.

dimitarvdimitrov commented:
Added a changelog entry. Also opened https://github.com/grafana/helm-charts/issues/1426 to start the discussion on removing the name overrides. Are you ok to merge this as-is? I added a changelog entry and I follow up based on the outcome of 1426.
